lib/constants.h

```diff
@@ -11,7 +11,7 @@ extern "C" {
 
 #include <snowflake/basic_types.h>
 #include "snowflake/platform.h"
-
+#define MAX_RETRY_COUNT 20
```

Comment:
Two ways to break away from the retry strategy 
1. Max retry count
2. Set max timeout and when the total elapsed time during the retries is >= timeout value then break out of the loop
first way constant number of retries
second way constant amount of retry time
I chose max retry count